fsspec/tests/conftest.py

```diff
@@ -169,13 +181,13 @@ def do_HEAD(self):
 
 @contextlib.contextmanager
 def serve():
-    server_address = ("", port)
+    server_address = ("", 0)
     httpd = HTTPServer(server_address, HTTPTestHandler)
     th = threading.Thread(target=httpd.serve_forever)
     th.daemon = True
     th.start()
     try:
-        yield f"http://127.0.0.1:{port}"
+        yield f"http://127.0.0.1:{httpd.server_port}"
```

Comment:
Since we are choosing to become port agnostic, this may be a good time to also become server address agnostic. `httpd.server_name` should be populated here, and might be other than the standard loopback, e.g., for systems that are IPv6 only.